### ISSUE TITLE
Algolia searcher: search runs twice causing Speak messages to not work in VoiceOver

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#320]
Seems the Algolia search runs twice (or the component is rendered twice?). To reproduce:
- in `composites/AlgoliaSearch/AlgoliaSearcher.js` put some console log at the beginning of `getSearchResults()`
- enter a term in the search
- observe your browser's console:

![screen shot 2017-09-29 at 10 49 11](https://user-images.githubusercontent.com/1682452/31008294-e55ddfaa-a503-11e7-9f54-1734e1ee3a0f.png)

Apart from efficiency considerations, this has a direct impact on the way `speak` works  and the Safari/VoiceOver bug with identical, repeated strings, see https://github.com/Yoast/a11y-speak/issues/11

The workaround for that bug is to add and remove a space `"\u00A0"` to each message to force VoiceOver think it's a different string, see https://github.com/Yoast/a11y-speak/pull/12

However, since the search runs twice (or the component is rendered twice), the space is adder/removed so fast that the workaround doesn't work any longer.

To reproduce:
- type a non-existing term search to get no results
- VoiceOver announces "No results found."

![screen shot 2017-09-29 at 10 54 45](https://user-images.githubusercontent.com/1682452/31008539-cd75dde2-a504-11e7-9318-b9970d48e1a1.png)

- add some more nonsense characters
- VoiceOver should announce again "No results found." but nothing happens

![screen shot 2017-09-29 at 10 54 53](https://user-images.githubusercontent.com/1682452/31008561-e5739ed4-a504-11e7-9232-52128a504dcb.png)

Observing the dom inspector in your browser, you may notice the space is actually added/removed but that's so fast that is barely noticeable.
